app/scripts/metamask-controller.test.js

```diff
@@ -836,63 +836,102 @@ describe('MetaMaskController', () => {
         ).toStrictEqual([]);
       });
 
-      it('returns the eth accounts from the CAIP-25 permission', async () => {
-        jest
-          .spyOn(metamaskController.permissionController, 'getCaveat')
-          .mockReturnValue({
-            value: {
-              requiredScopes: {},
-              optionalScopes: {
-                'eip155:1': {
-                  accounts: ['eip155:1:0xdead', 'eip155:1:0xbeef'],
+      describe('the wallet is locked', () => {
+        beforeEach(() => {
+          jest.spyOn(metamaskController, 'isUnlocked').mockReturnValue(false);
+        });
+
+        it('returns empty array if there is a CAIP-25 permission for the origin and ignoreLock is false', async () => {
+          jest
+            .spyOn(metamaskController.permissionController, 'getCaveat')
+            .mockReturnValue({
+              value: {
+                requiredScopes: {},
+                optionalScopes: {
+                  'eip155:1': {
+                    accounts: ['eip155:1:0xdead', 'eip155:1:0xbeef'],
+                  },
                 },
               },
-            },
-          });
-
-        expect(
-          metamaskController.getPermittedAccounts('test.com'),
-        ).toStrictEqual(['0xdead', '0xbeef']);
-      });
-    });
+            });
 
-    describe('#getPermittedAccountsSorted', () => {
-      it('gets the permitted accounts for the origin', async () => {
-        jest
-          .spyOn(metamaskController, 'getPermittedAccounts')
-          .mockReturnValue([]);
+          expect(
+            metamaskController.getPermittedAccounts('test.com', false),
+          ).toStrictEqual([]);
+        });
 
-        await metamaskController.getPermittedAccountsSorted('test.com');
+        it('returns some accounts if there is a CAIP-25 permission for the origin and ignoreLock is true', async () => {
```

Comment:
total nit, wording here just suggests you don't get all the accounts 😓 
```suggestion
        it('returns accounts if there is a CAIP-25 permission for the origin and ignoreLock is true', async () => {
```